pkg/registry/deployment/strategy.go

```diff
@@ -70,6 +72,12 @@ func (deploymentStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newDeployment := obj.(*extensions.Deployment)
 	oldDeployment := old.(*extensions.Deployment)
 	newDeployment.Status = oldDeployment.Status
+
+	if !reflect.DeepEqual(newDeployment.Spec, oldDeployment.Spec) ||
+		!reflect.DeepEqual(newDeployment.Labels, oldDeployment.Labels) ||
+		!reflect.DeepEqual(newDeployment.Annotations, oldDeployment.Annotations) {
```

Comment:
Annotations are copied to RSes. A Deployment's labels don't affect what it does. Please omit labels here and add a comment about why we just dump generation on spec and annotation changes.